eox_core/api/v1/permissions.py

```diff
@@ -18,14 +19,16 @@ def load_permissions():
     if settings.EOX_CORE_LOAD_PERMISSIONS:
         try:
             content_type = ContentType.objects.get_for_model(User)
-            obj, created = Permission.objects.get_or_create(  # pylint: disable=unused-variable
+            Permission.objects.get_or_create(
                 codename='can_call_eox_core',
                 name='Can access eox-core API',
                 content_type=content_type,
             )
-        except ProgrammingError:
-            # This code runs when the app is loaded, if a migration has not been done a ProgrammingError exception is raised
-            # we are bypassing those cases to let migrations run smoothly.
+        except (ProgrammingError, ImproperlyConfigured):
+            # This code runs when the app is loaded, if a migration has not been done a
+            # ProgrammingError or ImproperlyConfugured exception is raised. We are
+            # bypassing those cases to let migrations run smoothly. This error occurs
+            # when building the Open edX image.
```

Comment:
This might be a question for another time, but why create permissions in the __init__ file instead of the apps.py `ready` method, where we know everything is already loaded?

Reply:
I'm almost positive I tried adding it in the `ready` method instead of the `init` file for this specific plugin and it didn't work. I think it was because the ready method is called when the application registry is ready, and this happens before all migrations are applied (??), I'm not really sure. We could try it again in a different issue to see if it works

Reply:
Yes, from what I see it seems that doing it in the `__init__.py` or from `ready()` the same error occurs in the build time, I'm not sure why either.

But, where should we create the permissions? I think using the `ready()` method would be best.

Reply:
Mhmm I would say that If it's always raising the error now during the build, then I guess it's not relevant to have it in the `__init__ ` file anymore. It seems that it makes more sense and is better practice to have it in the `ready()` method.

@BryanttV, I'm guessing that installing eox-core once the environment is up directly on the platform does not raise any errors, right? Only during build time? If so, we would need to make sure that loading the permissions in the ready() method does not raise any errors either when installing the requirement manually

Reply:
> I'm guessing that installing eox-core once the environment is up directly on the platform does not raise any errors, right? Only during build time? 

Yes, only in build time

> If so, we would need to make sure that loading the permissions in the ready() method does not raise any errors either when installing the requirement manually

Yes, we could check all that in another issue.